.github/workflows/ossar.yml

```diff
@@ -0,0 +1,56 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# This workflow integrates a collection of open source static analysis tools
+# with GitHub code scanning. For documentation, or to provide feedback, visit
+# https://github.com/github/ossar-action
+name: OSSAR
+
+on:
+  push:
+    branches: [ "master", "stable" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
+  schedule:
+    - cron: '39 1 * * 0'
+
+permissions:
+  contents: read
+
+jobs:
+  OSSAR-Scan:
+    # OSSAR runs on windows-latest.
+    # ubuntu-latest and macos-latest support coming soon
+    permissions:
+      contents: read # for actions/checkout to fetch code
+      security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
+      actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
+    runs-on: windows-latest
```

Comment:
1. Regarding your suggestion:
> **Permissions Configuration Review**
> 
> Permissions are appropriately scoped to minimal requirements, enhancing security. Consider removing the duplicate `contents: read` permission at the job level if it's already provided at the workflow level.

That would violate the security principle of "Defence in Depth". I reject the suggestion on principle.

2. Please suppress suggestions to remove a single duplicate security control when it would violate the security principle of "Defence in Depth", in the future. Of course you may continue to suggest de-duplication when appropriate, if it does not impact security controls.

3. For posterity, 
 * Please summarize the principle of "Defence in Depth" in a tone appropriate for the PR walkthrough.
 * Then summarize the principle of "Defence in Depth" the tone of an angsty teenager.